### PR TITLE
[expo-dev-launcher] Fix bundled images

### PR DESCRIPTION
**Pull request by fson**
# Why

The images included in the `expo-dev-launcher` library didn't work previously.

The reason was that the JS bundle for dev launcher was included in the app in an unusual way, involving JS embedded inside a C++ source file and faking the value of `sourceURLForBridge`:
https://github.com/expo/expo/blob/e0bb254c889734f2ec6c7b688167f013587ed201/packages/expo-dev-launcher/ios/EXDevLauncherController.m#L32
React Native [resolves the asset URL relative to the JS bundle URL](https://github.com/expo/react-native/blob/85f6d8f2b00f72ee18cedef2052471ae16c34e26/Libraries/Image/AssetSourceResolver.js#L111-L123), so it wasn't able to locate the images, even when they were appropriately included in the resource bundle.

# How

- Instead of `EXDevLauncherBundle.cpp` load JS source from `main.jsbundle` included inside `EXDevLauncher.bundle` resource bundle, which makes loading it more similar to how it works in a bare React Native app. This makes the images resolve correctly.
- Add `resource_bundles` setting to the podspec definition. The resource bundle includes the JS bundle and the assets.
- Make build script output the JS bundle and assets in a folder structure expected by React Native so everything works as expected.

# Test Plan

Ensured the CLI icon was visible in the "Recently in Development" project items in the app, even when dev launcher bundle was not served with Metro but bundled in the Cocoapod. (Tested using BareExpo.)

# Checklist

N/A

### CODE REVIEW

File: packages/expo-dev-launcher/expo-dev-launcher.podspec [hunk 1]
```diff
@@ -18,6 +18,10 @@ Pod::Spec.new do |s|
   s.requires_arc   = true
   s.header_dir     = 'EXDevLauncher'
 
+  s.resource_bundles = {
+    'EXDevLauncher' => ['ios/assets', 'ios/main.jsbundle']
```
Comment (lukmccall) [inline packages/expo-dev-launcher/expo-dev-launcher.podspec:22]: I think it'll be better to create a subfolder for those assets and bundle.

**Review by lukmccall — COMMENTED** [review]
Great job 🚀

Did you test the Android? I think the same problem occurs on this platform too.

File: packages/expo-dev-launcher/expo-dev-launcher.podspec (hunk 1 above)
Comment (fson) [reply packages/expo-dev-launcher/expo-dev-launcher.podspec:22]: The way things are set up now the assets will already be in their own subfolder, so this seems fine to me, with no risk of collisions:
Screen Shot 2021-04-22 at 22 34 46

Comment (fson) [reply packages/expo-dev-launcher/expo-dev-launcher.podspec:22]: Or did you perhaps mean a new subfolder inside the `packages/expo-dev-launcher/ios` subfolder? What should it be named?

**Comment by fson** [comment]
I haven't tested on Android yet, but I could do that on Monday.

File: packages/expo-dev-launcher/expo-dev-launcher.podspec (hunk 1 above)
Comment (lukmccall) [reply packages/expo-dev-launcher/expo-dev-launcher.podspec:22]: I meant a new subfolder inside the `packages/expo-dev-launcher/ios` directory. We can call it `resources` for example.

**Review by lukmccall — APPROVED** [review]
I think we can merge that 🚀🚀🚀

**Pull request merged**